File: spring-boot-backend/src/main/java/com/aasha/web/entity/Incident.java
package com.aasha.web.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name = "incidents")
public class Incident {
    @Id
    @Column(length = 64)
    private String id;
    @Column(nullable = false)
    private String name;
    @Column(columnDefinition = "TEXT")
    private String description;
    @Column(nullable = false)
    private boolean active = true;
    @Column(nullable = false)
    private boolean searchable = true;
    private String district;
    private String state;
    private Double latitude;
    private Double longitude;
    @Column(name = "radius_km")
    private Double radiusKm;
    @Column(name = "severity", length = 20)
    private String severity;
    @Column(name = "created_by")
    private String createdBy;
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    void onCreate() {
        if (createdAt == null) createdAt = LocalDateTime.now();
        updatedAt = createdAt;
    }

    @PreUpdate
    void onUpdate() { updatedAt = LocalDateTime.now(); }

    public String getId() { return id; }
    public void setId(String value) { id = value; }
    public String getName() { return name; }
    public void setName(String value) { name = value; }
    public String getDescription() { return description; }
    public void setDescription(String value) { description = value; }
    public boolean isActive() { return active; }
    public void setActive(boolean value) { active = value; }
    public boolean isSearchable() { return searchable; }
    public void setSearchable(boolean value) { searchable = value; }
    public String getDistrict() { return district; }
    public void setDistrict(String value) { district = value; }
    public String getState() { return state; }
    public void setState(String value) { state = value; }
    public Double getLatitude() { return latitude; }
    public void setLatitude(Double value) { latitude = value; }
    public Double getLongitude() { return longitude; }
    public void setLongitude(Double value) { longitude = value; }
    public Double getRadiusKm() { return radiusKm; }
    public void setRadiusKm(Double value) { radiusKm = value; }
    public String getSeverity() { return severity; }
    public void setSeverity(String value) { severity = value; }
    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String value) { createdBy = value; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime value) { createdAt = value; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime value) { updatedAt = value; }
}

File: flutter-app/lib/features/incidents/data/models/incident.dart
class Incident {
  const Incident({
    required this.id,
    required this.name,
    this.description,
    this.active = true,
    this.searchable = true,
    this.district,
    this.state,
    this.latitude,
    this.longitude,
    this.radiusKm,
    this.severity,
    this.redZoneKm,
    this.yellowZoneKm,
    this.greenZoneKm,
  });

  final String id;
  final String name;
  final String? description;
  final bool active;
  final bool searchable;
  final String? district;
  final String? state;
  final double? latitude;
  final double? longitude;
  final double? radiusKm;
  final String? severity;
  final double? redZoneKm;
  final double? yellowZoneKm;
  final double? greenZoneKm;

  factory Incident.fromJson(Map<String, dynamic> json) => Incident(
        id: (json['id'] ?? '').toString(),
        name: (json['name'] ?? '').toString(),
        description: json['description'] as String?,
        active: json['active'] != false,
        searchable: json['searchable'] != false,
        district: json['district'] as String?,
        state: json['state'] as String?,
        latitude: (json['latitude'] as num?)?.toDouble(),
        longitude: (json['longitude'] as num?)?.toDouble(),
        radiusKm: (json['radiusKm'] ?? json['radius_km'] as num?)?.toDouble(),
        severity: json['severity'] as String?,
        redZoneKm: (json['redZoneKm'] ?? json['red_zone_km'] as num?)?.toDouble(),
        yellowZoneKm: (json['yellowZoneKm'] ?? json['yellow_zone_km'] as num?)?.toDouble(),
        greenZoneKm: (json['greenZoneKm'] ?? json['green_zone_km'] as num?)?.toDouble(),
      );
}
